[BUG]  test_storage_rewrite_detect_inplace_toggle unnecessarily triggers NVCC device compilation

### Required prerequisites

- [x] I have read the documentation <https://tilelang.com>.
- [x] I have searched the [Issue Tracker](https://github.com/tile-ai/tilelang/issues) that this hasn't already been reported. (comment there if it has.)

### What version of TileLang are you using?

0.1.10

### System information

Install from source
GPU driver CUDA capability: 13.2
NVIDIA driver: 596.49

PyTorch: 2.12.0+cu130
torch.version.cuda: 13.0

Python CUDA packages:
cuda-toolkit: 13.0.2
nvidia-cuda-nvcc: 13.3.33
nvidia-cuda-cccl: 13.3.3.3.1
nvidia-cuda-nvrtc: 13.0.88
nvidia-cuda-runtime: 13.0.96

### Problem description

`testing/python/components/test_storage_rewrite_detect_inplace.py::test_storage_rewrite_detect_inplace_toggle` is intended to validate CUDA source codegen for `TL_STORAGE_REWRITE_DETECT_INPLACE`.

However, the test gets the kernel source through the default JIT path:
```python
kernel = _compile_kernel_without_inplace()
source = kernel.get_kernel_source()
```

With the default tvm_ffi backend, this path triggers full NVCC device compilation before returning the source. As a result, the test can fail due to CUDA compiler/toolkit/header mismatch, even though the test only needs generated CUDA source text.

This makes the test environment-dependent and can mask the actual storage rewrite behavior being tested.

### Reproducible example code

`python -m pytest testing/python/components/test_storage_rewrite_detect_inplace.py -q`


### Traceback

```pytb
.../tilelang/.venv/lib/python3.12/site-packages/nvidia/cu13/include/cccl/cuda/std/__cccl/cuda_toolkit.h:41:8: error: #error "CUDA compiler and CUDA toolkit headers are incompatible, please check your include paths"
   41 | #      error "CUDA compiler and CUDA toolkit headers are incompatible, please check your include paths"
      |        ^~~~~
```

### Expected behavior

The test should only generate CUDA source code and assert the effect of `TL_STORAGE_REWRITE_DETECT_INPLACE` on storage rewrite output.

It should not require successful NVCC cubin/device compilation, because the test only checks generated source strings such as `read = (read * 2);` and `write = (read * 2);`.

A CUDA compiler/toolkit mismatch should not cause this source-codegen regression test to fail before reaching its assertions.


### Additional context

_No response_

--- a/testing/python/components/test_storage_rewrite_detect_inplace.py
+++ b/testing/python/components/test_storage_rewrite_detect_inplace.py
@@ -1,6 +1,6 @@
 import tilelang
 import tilelang.testing
-from tilelang import language as T
+from tilelang import language as T, tvm
 
 
 @tilelang.jit
@@ -42,12 +42,13 @@ def _compile_kernel_with_inplace():
 
 
 def _get_device_kernel_script(detect_inplace: bool) -> str:
+    kernel = _compile_kernel_with_inplace if detect_inplace else _compile_kernel_without_inplace
+    pass_configs = {}
     if detect_inplace:
-        kernel = _compile_kernel_with_inplace()
-    else:
-        kernel = _compile_kernel_without_inplace()
-    source = kernel.get_kernel_source()
-    return source
+        pass_configs[tilelang.PassConfigKey.TL_STORAGE_REWRITE_DETECT_INPLACE] = True
+    with tvm.transform.PassContext(config=pass_configs):
+        artifact = tilelang.lower(kernel.get_tir(), target="cuda")
+    return artifact.kernel_source
 
 
 def test_storage_rewrite_detect_inplace_toggle():

